calendar/engine/subscription_test.go

```diff
@@ -0,0 +1,460 @@
+				require.Equal(t, expectedSubscription, sub)
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.setupMock()
+
+			sub, err := mscalendar.LoadMyEventSubscription()
+
+			tt.assertion(sub, err)
+		})
+	}
+}
+
+func TestListRemoteSubscriptions(t *testing.T) {
+	mscalendar, mockStore, _, _, _, mockClient, _ := MockSetup(t)
+	mockUser := GetMockUser()
+
+	tests := []struct {
+		name      string
+		setupMock func()
+		assertion func(subs []*remote.Subscription, err error)
+	}{
+		{
+			name: "error filtering with user",
+			setupMock: func() {
+				mockUser.User = nil
+				mscalendar.client = nil
+				mscalendar.actingUser = &User{MattermostUserID: "testActingUserID"}
+				mockStore.EXPECT().LoadUser("testActingUserID").Return(nil, errors.New("error filtering user")).Times(1)
+			},
+			assertion: func(subs []*remote.Subscription, err error) {
+				require.Error(t, err)
+				require.EqualError(t, err, "error withClient in ListRemoteSubscriptions: It looks like your Mattermost account is not connected to testDisplayName. Please connect your account using `/testCommandTrigger connect`.: error filtering user")
+			},
+		},
+		{
+			name: "error listing subscription",
+			setupMock: func() {
+				mockUser.User = &store.User{Settings: store.Settings{}, Remote: &remote.User{ID: "testRemoteID"}}
+				mscalendar.client = mockClient
+				mscalendar.actingUser = &User{User: &store.User{Remote: &remote.User{ID: "testActingUserRemoteID"}}, MattermostUserID: "testActingUserID"}
+				mscalendar.actingUser.Settings.EventSubscriptionID = "testEventSubscriptionID"
+				mockClient.EXPECT().ListSubscriptions().Return(nil, errors.New("error listing subscriptions"))
+			},
+			assertion: func(subs []*remote.Subscription, err error) {
+				require.Error(t, err)
+				require.EqualError(t, err, "error listing subscriptions")
+			},
+		},
+		{
+			name: "subscriptions listed successfully",
+			setupMock: func() {
+				mockUser.User = &store.User{Settings: store.Settings{}, Remote: &remote.User{ID: "testRemoteID"}}
+				mscalendar.client = mockClient
+				mscalendar.actingUser = &User{User: &store.User{Remote: &remote.User{ID: "testActingUserRemoteID"}}, MattermostUserID: "testActingUserID"}
+				mscalendar.actingUser.Settings.EventSubscriptionID = "testEventSubscriptionID"
+				mockClient.EXPECT().ListSubscriptions().Return([]*remote.Subscription{{ID: "sub1"}, {ID: "sub2"}}, nil)
+			},
+			assertion: func(subs []*remote.Subscription, err error) {
+				require.NoError(t, err)
+				require.Equal(t, []*remote.Subscription{{ID: "sub1"}, {ID: "sub2"}}, subs)
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.setupMock()
+
+			subs, err := mscalendar.ListRemoteSubscriptions()
+
+			tt.assertion(subs, err)
+		})
+	}
+}
+
+func TestRenewMyEventSubscription(t *testing.T) {
+	mscalendar, mockStore, _, mockRemote, mockPluginAPI, mockClient, _ := MockSetup(t)
+	mockUser := GetMockUser()
```

Comment:
We can create two mock utils one returning with mockUser.User = nil and another with value.

Reply:
Also, make a note, once this PR is merged we should use this approach in the previous testcases as well in a separate PR